src/main/kotlin/com/avail/AvailTask.kt

```diff
@@ -0,0 +1,256 @@
+
+/**
+ * An `AvailTask` extends [Runnable] with a priority. Instances are intended to
+ * be executed only by [Avail&#32;threads][AvailThread].
+ *
+ * @author Todd L Smith &lt;todd@availlang.org&gt;
+ *
+ * @property
+ *      The priority of the [task][AvailTask].  It must be a value in the range
+ *      0..255. @see [quasiDeadline]
+ * @constructor
+ * Construct a new `AvailTask`.
+ *
+ * @param priority
+ *   The desired priority, a long tied to milliseconds since the current epoch.
+ * @param body
+ *   The action to execute for this task.
+ */
+class AvailTask constructor(
+	val priority: Int,
+	body: Function0<Unit>) : Comparable<AvailTask>, Runnable
+{
+	/** The action to perform for this task.  */
+	private val body: Function0<Unit>
+
+
+	/**
+	 * The quasi-deadline of the task.  This is the moment that the task can no
+	 * longer be preceded by a new task.  Given a priority in the range 0..255,
+	 * the quasi-deadline is System.currentTimeMillis() plus a time delta.  The
+	 * delta is 1000ms * (1 - (priority / 256)).  That places it one second in
+	 * the future for priority = 0, and places it essentially at the current
+	 * moment for priority = 255.
+	 */
+	private val quasiDeadline: Long
+
+	// Note that using Long.compare(long) would deal with counter overflow
+	// in a way we don't want.  Instead, assume a task doesn't sit in the
+	// queue for a century or two(!), and use a technically non-transitive
+	// comparison operation, checking if the difference is negative.
+	// noinspection SubtractionInCompareTo
+	// TODO confirm comment still valid
+	override fun compareTo(other: AvailTask): Int =
+		(quasiDeadline - other.quasiDeadline).compareTo(0)
+
+	override fun run()
+	{
+		try
+		{
+			body.invoke()
+		}
+		catch (e: Throwable)
+		{
+			// Report the exception immediately, then suppress the error.  If we
+			// allowed the error to propagate, it would cause an Interpreter to
+			// be silently lost, and the next creation could overflow the
+			// maximum number of Interpreters.
+			System.err.println("Unexpected internal failure in AvailTask:\n")
+			e.printStackTrace()
+		}
+	}
+
+	companion object
+	{
+		/**
+		 * Answer an `AvailTask` suitable for resuming the specified
+		 * [fiber][FiberDescriptor] using the specified action. If the
+		 * continuation fails for any reason, then it
+		 * [aborts][Interpreter.abortFiber] the fiber and invokes the [failure
+		 * continuation][AvailObject.failureContinuation] with the terminal
+		 * [throwable][Throwable].
+		 *
+		 * @param fiber
+		 *   A fiber.
+		 * @param body
+		 *   What to do to resume execution of the fiber.
+		 * @return
+		 *   An action that sets the execution state of the fiber to
+		 *   [running][ExecutionState.RUNNING], binds it to the running
+		 *   [thread][AvailThread]'s [interpreter][Interpreter], and then runs
+		 *   the specified continuation.
+		 */
+		fun forFiberResumption(
+			fiber: A_Fiber,
+			body: Function0<Unit>): Function0<Unit>
```

Comment:
Please globally replace all usage of the pattern Function\d within *.kt files with Kotlin's (X)->Y notation.